ProcessLib/GroundwaterFlow/GroundwaterFlowProcess.h

```diff
@@ -34,14 +38,55 @@ class GroundwaterFlowProcess final : public Process
             process_variables,
         GroundwaterFlowProcessData&& process_data,
         SecondaryVariableCollection&& secondary_variables,
-        NumLib::NamedFunctionCaller&& named_function_caller);
+        NumLib::NamedFunctionCaller&& named_function_caller,
+        MeshLib::Mesh* balance_mesh, std::string&& balance_pv_name,
+        std::string&& balance_out_fname);
 
     //! \name ODESystem interface
     //! @{
 
     bool isLinear() const override { return true; }
     //! @}
 
+    std::vector<double> getFlux(std::size_t element_id,
+                                MathLib::Point3d const& p,
+                                GlobalVector const& x) const override
+    {
+        // fetch local_x from primary variable
+        std::vector<GlobalIndexType> indices_cache;
+        auto const r_c_indices = NumLib::getRowColumnIndices(
+            element_id, *_local_to_global_index_map, indices_cache);
+        std::vector<double> local_x(x.get(r_c_indices.rows));
+
+        return _local_assemblers[element_id]->getFlux(p, local_x);
+    }
+
+    void postTimestepConcreteProcess(GlobalVector const& x) override
+    {
+        if (_balance_mesh) // computing the balance is optional
+        {
+            std::vector<double> init_values(
+                _balance_mesh->getNumberOfElements(), 0.0);
+            MeshLib::addPropertyToMesh(*_balance_mesh, _balance_pv_name,
```

Comment:
Obviously it has to be configurable at which timesteps the surface fluxes will be computed.
I suggest to implement it in a follow-up PR, and to leave the current implementation as it is.